src/librustc/mir/interpret/allocation.rs

```diff
@@ -688,6 +693,56 @@ impl<Tag> DerefMut for Relocations<Tag> {
     }
 }
 
+/// A partial, owned list of relocations to transfer into another allocation.
+pub struct AllocationRelocations<Tag> {
+    relative_relocations: Vec<(Size, (Tag, AllocId))>,
+}
+
+impl<Tag: Copy, Extra> Allocation<Tag, Extra> {
+    pub fn prepare_relocation_copy(
+        &self,
+        cx: &impl HasDataLayout,
+        src: Pointer<Tag>,
+        size: Size,
+        dest: Pointer<Tag>,
+        length: u64,
+    ) -> AllocationRelocations<Tag> {
```

Comment:
What about `get_relocations_range` as name?

Reply:
The return value is destination specific since it already offsets correctly. I didn't think `get` would make this sufficiently clear. It would be possible to move the final offsetting with `dest.offset into `mark_relocation_range` but I didn't want to modify it to stay as close to the current code as possible.

Reply:
Hm, makes sense.